btsieve/src/routes.rs

```diff
@@ -162,14 +164,26 @@ pub fn delete_query<
 }
 
 #[allow(clippy::needless_pass_by_value)]
-pub fn get_or_create_query<Q: Send + Eq, QR: QueryRepository<Q>, C: 'static + Send + Sync>(
+pub fn get_or_create_query<
+    Q: Send + Sync + Eq + Clone + 'static,
+    QR: QueryRepository<Q>,
+    QRR: QueryResultRepository<Q>,
+    C: 'static + Send + Sync,
+    E: 'static,
+    BS: MatchingTransactions<Q, Error = E>,
+>(
     _client: Arc<C>,
     _network: String,
-    query_repository: Arc<QR>,
+    blocksource: Arc<BS>,
+    active_query_repository: Arc<QR>,
```

Comment:
:question: Why did the name change here? I don't mind adding the active, but I'd add it everywhere. Otherwise it can get confusing.